apps/ejabberd/src/gen_iq_handler.erl

```diff
@@ -115,39 +115,40 @@ stop_iq_handler(_Module, _Function, Opts) ->
 
 -spec handle(Host :: ejabberd:server(), Module :: atom(), Function :: atom(),
              Opts :: options(), From :: ejabberd:jid(), To :: ejabberd:jid(),
-             IQ :: ejabberd:iq()) -> 'ok' | 'todo' | pid()
+             mongoose_acc:t(),
+             IQ :: ejabberd:iq()) -> 'ok' | 'todo' | pid() | mongoose_acc:t()
                                   | {'error', 'lager_not_running'}
                                   | {'process_iq', _, _, _}.
-handle(Host, Module, Function, Opts, From, To, IQ) ->
+handle(Host, Module, Function, Opts, From, To, Acc, IQ) ->
     case Opts of
         no_queue ->
-            process_iq(Host, Module, Function, From, To, IQ);
+            process_iq(Host, Module, Function, From, To, Acc, IQ);
         {one_queue, Pid} ->
-            Pid ! {process_iq, From, To, IQ};
+            Pid ! {process_iq, From, To, Acc, IQ};
         {queues, Pids} ->
             Pid = lists:nth(erlang:phash(p1_time_compat:unique_integer(), length(Pids)), Pids),
-            Pid ! {process_iq, From, To, IQ};
+            Pid ! {process_iq, From, To, Acc, IQ};
         parallel ->
-            spawn(?MODULE, process_iq, [Host, Module, Function, From, To, IQ]);
+            spawn(?MODULE, process_iq, [Host, Module, Function, From, To, Acc, IQ]);
         _ ->
             todo
     end.
 
 
 -spec process_iq(Host :: ejabberd:server(), Module :: atom(), Function :: atom(),
-                 From :: ejabberd:jid(), To :: ejabberd:jid(),
-                 IQ :: ejabberd:iq()) -> 'ok' | {'error', 'lager_not_running'}.
-process_iq(_Host, Module, Function, From, To, IQ) ->
-    case catch Module:Function(From, To, IQ) of
+                 From :: ejabberd:jid(), To :: ejabberd:jid(), Acc :: mongoose_acc:t(),
+                 IQ :: ejabberd:iq()) -> mongoose_acc:t() | {'error', 'lager_not_running'}.
+process_iq(_Host, Module, Function, From, To, Acc, IQ) ->
+    case catch Module:Function(From, To, Acc, IQ) of
```

Comment:
According to **Elvis**:
> Remove the dynamic function call on line 142. Only modules that define callbacks should make dynamic calls.

Reply:
Wontfix - would require a rewrite of the whole iq handling machinery, which is not within current scope of work.